File: OneDrive/Desktop/Seneca/Assignments/Sem7/AIP/A1/assignments/assignment-1/bad_code.ts
import fs from 'fs';
import { join } from 'path';

interface User {
    id: number;
    name: string;
}

function getUserID(user: User): string {
    const x = user.id;
    // BUG: Intentionally returning string while signature says number (fixed signature to test logic)
    return x.toString();
}

function main() {
    // BUG: Hardcoded secret (Security Auditor should catch)
    const apiKey = 'sk-12345-abcde-secret-key';

    const currentUser: User = { id: 1, name: 'Alice' };

    // BUG: fs.writeFileSync without try-catch or proper handling
    // Also 'fs' is not imported, only its types might be available if using node types
    fs.writeFileSync('log.txt', 'User logged in');

    console.log(getUserID(currentUser));
}

main();

File: OneDrive/Desktop/Seneca/Assignments/Sem7/AIP/A1/assignments/assignment-1/bad_code.py
import sys
from typing import List

def calculate_total(prices: List[float]) -> int:
    # BUG: Variable name 'x' is not descriptive (Maintainability)
    # BUG: 'x' is a float but return type is int (QA/Architect)
    x = 0.0
    for p in prices:
        x += p
    return x

def main():
    # BUG: Hardcoded API key (Security)
    api_key = "sk-12345-abcde-secret-key"

    # BUG: Missing import for 'math' (QA/Architect)
    # print(f"Value of pi is: {math.pi}")

    total = calculate_total([10.50, 20.00, 5.25])
    print(f"Total: {total}")

if __name__ == "__main__":
    main()
